src/vm/internal/otable_test.cc

```diff
@@ -0,0 +1,48 @@
+#define TESTING
+
+#include<string>
+struct object_t
+{
+public:
+    std::string obname;
+
+    object_t(std::string const & s)
+    :obname(s) {}
+};
+
+#include "otable.cc"
```

Comment:
once again, this is not how unit-test should work,  you can't include ".cc" file in here, you have to link with the object file. You have to figure out exactly how, but current way is definitely wrong.

Reply:
I actually am quite sure this isn't the canonical way either. I am however not sure it's "wrong". The problem is currently given how the testing works I need to at the very least pass in a define flag to the object file so at present this needs to be set somewhere. I think I have thought of a workaround for this by using the -I flag to point at a different object_t using a compiler flag. That way I can avoid the problems including all the forward declared functions in the object.h which are irrelevant for testing the code.